src/charm.py

```diff
@@ -1,104 +1,402 @@
 #!/usr/bin/env python3
 # Copyright 2021 Canonical Ltd.
 # See LICENSE file for licensing details.
-#
-# Learn more at: https://juju.is/docs/sdk
 
-"""Charm the service.
-
-Refer to the following post for a quick-start guide that will help you
-develop a new k8s charm using the Operator Framework:
-
-    https://discourse.charmhub.io/t/4208
-"""
+"""Charmed PgBouncer connection pooler."""
```

Comment:
Some parts of the code are a little confusing to me (since this a lot of this is new to me). I think some of it could use some comments before a code block when it isn't obvious whats going on. (IMHO) 

But if you disagree then that is also fine. I can leave comments where I got confused if you want to add comments (feel free to add all/some/none).

Reply:
Yes Mia! Clarifying comments in code is one of the hallmarks of good coding.  Also, please do not fall into the trap of translating to English what is written in code.